src/misc/rwqueue.cpp

```diff
@@ -74,6 +74,53 @@ void RWQueue<T>::Enqueue(T&& item)
 	has_items.notify_one();
 }
 
+template <typename T>
+void RWQueue<T>::BulkEnqueue(std::vector<T>& from_source, const size_t num_requested)
+{
+	// Check that the source can provide the quantity requested
+	assert(num_requested <= from_source.size());
+
+	// The source iterator will be incremented with each chunk enqueued
+	auto source_start = from_source.begin();
+
+	// The remaining count is decremented until no more items need enqueing
+	auto num_remaining = num_requested;
```

Comment:
I'd remove the above 3 line comments; they don't explain anything that isn't obvious from the code.

Reply:
Done!